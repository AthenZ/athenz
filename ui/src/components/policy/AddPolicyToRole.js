/*
 * Copyright The Athenz Authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React from 'react';
import AddModal from '../modal/AddModal';
import AddRuleForRoleForm from './AddRuleForRoleForm';
import RequestUtils from '../utils/RequestUtils';
import { addPolicy } from '../../redux/thunks/policies';
import { connect } from 'react-redux';

export class AddPolicyToRole extends React.Component {
    constructor(props) {
        super(props);
        this.onSubmit = this.onSubmit.bind(this);
        this.onChange = this.onChange.bind(this);
        this.state = {
            showModal: !!this.props.showAddPolicy,
            case: false,
        };
    }

    onSubmit() {
        if (!this.state.name || this.state.name === '') {
            this.setState({
                errorMessage: 'Policy name is required.',
            });
            return;
        }

        if (!this.state.action || this.state.action === '') {
            this.setState({
                errorMessage: 'Rule action is required.',
            });
            return;
        }

        if (!this.state.resource || this.state.resource === '') {
            this.setState({
                errorMessage: 'Rule resource is required.',
            });
            return;
        }

        this.props
            .addPolicy(
                this.props.domain,
                this.state.name,
                this.props.role,
                this.state.resource,
                this.state.action,
                this.state.effect,
                this.state.case,
                this.props._csrf
            )
            .then((data) => {
                this.setState({ showModal: false });
                this.props.onSubmit(`${this.state.name}`, false);
            })
            .catch((err) => {
                this.setState({
                    errorMessage: RequestUtils.xhrErrorCheckHelper(err),
                });
            });
    }

    onChange(key, value) {
        this.setState({ [key]: value });
    }

    render() {
        return (
            <AddModal
                isOpen={this.state.showModal}
                cancel={this.props.onCancel}
                submit={this.onSubmit}
                title={`Add Policy to ${this.props.domain}`}
                errorMessage={this.state.errorMessage}
                sections={
                    <AddRuleForRoleForm
                        domain={this.props.domain}
                        onChange={this.onChange}
                        isPolicy={true}
                    />
                }
            />
        );
    }
}

const mapDispatchToProps = (dispatch) => ({
    addPolicy: (
        domain,
        policyName,
        role,
        resource,
        action,
        effect,
        caseSensitive,
        _csrf
    ) =>
        dispatch(
            addPolicy(
                domain,
                policyName,
                role,
                resource,
                action,
                effect,
                caseSensitive,
                _csrf
            )
        ),
});

export default connect(null, mapDispatchToProps)(AddPolicyToRole);
