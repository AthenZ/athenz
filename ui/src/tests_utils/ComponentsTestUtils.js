/*
 * Copyright The Athenz Authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { initStore } from '../redux/store';
import { getExpiryTime } from '../redux/utils';

export const renderWithRedux = (component, initialState = {}) => {
    return render(
        <Provider store={initStore(initialState)}>{component}</Provider>
    );
};

export const buildUserForState = (user) => {
    return {
        expiry: getExpiryTime(),
        user,
    };
};

export const getStateWithUser = (user, initialState = {}) => {
    return { ...initialState, user };
};

export const buildDomainDataForState = (domainData, domainName = 'dom') => {
    return {
        domainName,
        expiry: getExpiryTime(),
        domainData,
    };
};

export const getStateWithDomainData = (domainData, initialState = {}) => {
    return { ...initialState, domainData: domainData };
};

export const getStateWithRoles = (roles, initialState = {}) => {
    return { ...initialState, roles };
};

export const buildRolesForState = (roles, domainName = 'dom') => {
    return {
        domainName,
        expiry: getExpiryTime(),
        roles,
    };
};

export const buildPoliciesForState = (policies, domainName = 'dom') => {
    return {
        domainName,
        expiry: getExpiryTime(),
        policies,
    };
};

export const getStateWithPolicies = (policies, initialState = {}) => {
    return { ...initialState, policies };
};

export const buildServicesForState = (
    services,
    domainName = 'dom',
    allProviders = []
) => {
    return {
        domainName,
        expiry: getExpiryTime(),
        services,
        allProviders,
    };
};

export const getStateWithServices = (services, initialState = {}) => {
    return { ...initialState, services };
};

export const buildMicrosegmentationForState = (
    inboundOutboundList,
    domainName = 'dom'
) => {
    return {
        domainName,
        expiry: getExpiryTime(),
        inboundOutboundList,
    };
};

export const getStateWithMicrosegmentation = (
    microsegmentation,
    initialState = {}
) => {
    return { ...initialState, microsegmentation };
};

export const buildGroupsForState = (groups, domainName = 'dom') => {
    return {
        domainName,
        expiry: getExpiryTime(),
        groups,
    };
};

export const getStateWithGroups = (groups, initialState = {}) => {
    return { ...initialState, groups };
};

export const buildServiceDependencies = (
    serviceDependencies,
    domainName = 'dom'
) => {
    return {
        domainName,
        expiry: getExpiryTime(),
        serviceDependencies,
    };
};

export const getStateWithServiceDependencies = (
    serviceDependencies,
    initialState = {}
) => {
    return { ...initialState, serviceDependencies: serviceDependencies };
};

export const mockRolesApiCalls = () => {
    return {
        getRoles: jest.fn().mockReturnValue(
            new Promise((resolve, reject) => {
                resolve([]);
            })
        ),
        getRoleMembers: jest.fn().mockReturnValue(
            new Promise((resolve, reject) => {
                resolve({});
            })
        ),
    };
};

export const mockPoliciesApiCalls = () => {
    return {
        getPolicies: jest.fn().mockReturnValue(
            new Promise((resolve, reject) => {
                resolve([]);
            })
        ),
        getPolicy: jest.fn().mockReturnValue(
            new Promise((resolve, reject) => {
                resolve({});
            })
        ),
        // getRoleMembers: jest.fn().mockReturnValue(
        //     new Promise((resolve, reject) => {
        //         resolve({});
        //     })
        // ),
    };
};

export const mockAllDomainDataApiCalls = (domainDetails, headerDetails) => {
    return {
        getDomain: jest.fn().mockReturnValue(
            new Promise((resolve, reject) => {
                resolve(domainDetails || {});
            })
        ),
        isAWSTemplateApplied: jest.fn().mockReturnValue(
            new Promise((resolve, reject) => {
                resolve('');
            })
        ),
        getHeaderDetails: jest.fn().mockReturnValue(
            new Promise((resolve, reject) => {
                resolve(headerDetails || {});
            })
        ),
        getPendingDomainMembersListByDomain: jest.fn().mockReturnValue(
            new Promise((resolve, reject) => {
                resolve([]);
            })
        ),
        getFeatureFlag: jest.fn().mockReturnValue(
            new Promise((resolve, reject) => {
                resolve('');
            })
        ),
        getMeta: jest.fn().mockReturnValue(
            new Promise((resolve, reject) => {
                resolve([]);
            })
        ),
        getAuthorityAttributes: jest.fn().mockReturnValue(
            new Promise((resolve, reject) => {
                resolve({});
            })
        ),
        getReviewGroups: jest.fn().mockReturnValue([]),
        getReviewRoles: jest.fn().mockReturnValue([]),
        getPageFeatureFlag: jest.fn().mockResolvedValue({}),
    };
};

export const getStateWithUserList = (userList, initialState = {}) => {
    return { ...initialState, user: userList };
};

export const getStateWithDomainDataAndUserList = (
    domainData,
    userList,
    initialState = {}
) => {
    return { ...initialState, domainData: domainData, user: userList };
};
