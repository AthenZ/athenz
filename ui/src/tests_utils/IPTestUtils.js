/*
 * Copyright The Athenz Authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export default class IPTestUtils {
    static getRandomInt(min, max) {
        return Math.floor(Math.random() * (max - min + 1)) + min;
    }

    static generateRandomIP() {
        return `${IPTestUtils.getRandomInt(0, 255)}.${IPTestUtils.getRandomInt(
            0,
            255
        )}.${IPTestUtils.getRandomInt(0, 255)}.${IPTestUtils.getRandomInt(
            0,
            255
        )}`;
    }

    static generateRandomCIDR() {
        const ip = IPTestUtils.generateRandomIP();
        const prefix = IPTestUtils.getRandomInt(0, 32);
        return `${ip}/${prefix}`;
    }

    static generateRandomString(length) {
        const characters =
            'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
        let result = '';
        for (let i = 0; i < length; i++) {
            result += characters.charAt(
                Math.floor(Math.random() * characters.length)
            );
        }
        return result;
    }

    static generateRandomFQDN() {
        const subdomain = IPTestUtils.generateRandomString(
            IPTestUtils.getRandomInt(3, 10)
        );
        const domain = IPTestUtils.generateRandomString(
            IPTestUtils.getRandomInt(3, 10)
        );
        const tld = IPTestUtils.generateRandomString(
            IPTestUtils.getRandomInt(2, 6)
        ); // Top Level Domain, like com, net, org, etc.
        return `${subdomain}.${domain}.${tld}`;
    }
}
