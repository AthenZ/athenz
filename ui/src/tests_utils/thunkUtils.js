/*
 * Copyright The Athenz Authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

import {
    addAssertionPolicyVersionToStore,
    deleteAssertionPolicyVersionFromStore,
    loadPolicies,
} from '../redux/actions/policies';
import { getExpiryTime } from '../redux/utils';
import { storeDomainData, storePolicies } from '../redux/actions/domains';
import { loadDomainData } from '../redux/actions/domain-data';

export const getLoadDomainDataAction = (domainName, domainData) => {
    return loadDomainData(domainData, domainName, getExpiryTime());
};

export const getStoreDomainDataAction = (domainData) => {
    return storeDomainData(domainData);
};

export const getLoadPoliciesAction = (domainName, policies = {}) => {
    return loadPolicies(policies, domainName, getExpiryTime());
};

export const getStorePoliciesAction = (policiesData) => {
    return storePolicies(policiesData);
};

export const getDeleteAssertionPolicyVersionAction = (
    policyName,
    version,
    assertionId
) => {
    return deleteAssertionPolicyVersionFromStore(
        policyName,
        version,
        assertionId
    );
};

export const getAddAssertionPolicyVersionAction = (
    policyName,
    version,
    newAssertion
) => {
    return addAssertionPolicyVersionToStore(policyName, version, newAssertion);
};
