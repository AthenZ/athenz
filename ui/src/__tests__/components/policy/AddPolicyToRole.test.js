/*
 * Copyright The Athenz Authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import AddPolicyToRole from '../../../components/policy/AddPolicyToRole';
import { renderWithRedux } from '../../../tests_utils/ComponentsTestUtils';

describe('AddPolicyToRole', () => {
    it('should render', () => {
        const showAddPolicy = true;
        const cancel = function () {};
        const domain = 'domain';
        const role = 'roleName';
        const { getByTestId } = renderWithRedux(
            <AddPolicyToRole
                showAddPolicy={showAddPolicy}
                onCancel={cancel}
                domain={domain}
                role={role}
            />
        );
        const addPolicy = getByTestId('add-modal-message');
        expect(addPolicy).toMatchSnapshot();
    });

    it('should render fail to submit policy name is required', () => {
        const showAddPolicy = true;
        const cancel = function () {};
        const domain = 'domain';
        const role = 'roleName';
        const { getByTestId, getByText } = renderWithRedux(
            <AddPolicyToRole
                showAddPolicy={showAddPolicy}
                onCancel={cancel}
                domain={domain}
                role={role}
            />
        );
        const addPolicy = getByTestId('add-modal-message');
        expect(addPolicy).toMatchSnapshot();

        fireEvent.click(getByText('Submit'));

        expect(getByText('Policy name is required.')).not.toBeNull();
    });

    it('should render failed to submit action is required', () => {
        const showAddPolicy = true;
        const cancel = function () {};
        const domain = 'domain';
        const role = 'roleName';
        const { getByTestId, getByText, getByPlaceholderText } =
            renderWithRedux(
                <AddPolicyToRole
                    showAddPolicy={showAddPolicy}
                    onCancel={cancel}
                    domain={domain}
                    role={role}
                />
            );
        const addPolicy = getByTestId('add-modal-message');
        expect(addPolicy).toMatchSnapshot();

        fireEvent.change(getByPlaceholderText('Policy Name'), {
            target: { value: 'policyname' },
        });
        fireEvent.click(getByText('Submit'));

        expect(getByText('Rule action is required.')).not.toBeNull();
    });
});
