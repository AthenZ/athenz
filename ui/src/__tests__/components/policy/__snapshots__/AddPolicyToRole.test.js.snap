// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPolicyToRole should render 1`] = `
<div
  class="css-1j49gd8"
  data-testid="add-modal-message"
>
  <div
    autocomplete="off"
    class="css-vcd2ur"
    data-testid="add-rule-form"
  >
    <div
      class="css-y97ykc"
    >
      <label
        class="css-fc0mv0-input-label denali-input-label css-9xcde0"
        for="policy-name"
      >
        Policy Name
      </label>
      <div
        class="css-1xqce5"
      >
        <div
          class="css-dqae60-input denali-input animated css-fvi5z5"
          data-testid="input-wrapper"
        >
          <input
            autocomplete="off"
            data-testid="input-node"
            id="policy-name"
            name="policy-name"
            placeholder="Policy Name"
            value=""
          />
          <div
            class="message"
            data-testid="message"
          />
        </div>
      </div>
    </div>
    <div
      class="css-y97ykc"
    >
      <label
        class="css-fc0mv0-input-label denali-input-label css-9xcde0"
        for="rule-effect"
      >
        Rule Effect
      </label>
      <div
        class="css-1xqce5"
      >
        <div
          class="css-9o2u2w-button-group denali-radiobutton-group css-1ldveyy"
          data-testid="radiobuttongroup"
        >
          <div
            class="css-fe5os4-radiobutton denali-radiobutton"
            data-testid="radiobutton-wrapper"
          >
            <input
              checked=""
              id="radiobutton-rule-effect-radio-group-ALLOW"
              name="rule-effect-radio-group"
              type="radio"
              value="ALLOW"
            />
            <label
              for="radiobutton-rule-effect-radio-group-ALLOW"
            >
              Allow
            </label>
          </div>
          <div
            class="css-fe5os4-radiobutton denali-radiobutton"
            data-testid="radiobutton-wrapper"
          >
            <input
              id="radiobutton-rule-effect-radio-group-DENY"
              name="rule-effect-radio-group"
              type="radio"
              value="DENY"
            />
            <label
              for="radiobutton-rule-effect-radio-group-DENY"
            >
              Deny
            </label>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-y97ykc"
    >
      <label
        class="css-fc0mv0-input-label denali-input-label css-9xcde0"
        for="rule-action"
      >
        Rule Action
      </label>
      <div
        class="css-1xqce5"
      >
        <div
          class="css-dqae60-input denali-input animated css-fvi5z5"
          data-testid="input-wrapper"
        >
          <input
            autocomplete="off"
            data-testid="input-node"
            id="rule-action"
            name="rule-action"
            placeholder="Rule Action"
            value=""
          />
          <div
            class="message"
            data-testid="message"
          />
        </div>
      </div>
    </div>
    <div
      class="css-y97ykc"
    >
      <label
        class="css-fc0mv0-input-label denali-input-label css-9xcde0"
        for="rule-resource"
      >
        Rule Resource
      </label>
      <div
        class="css-1xqce5"
      >
        <div
          class="css-dqae60-input denali-input animated css-fvi5z5"
          data-testid="input-wrapper"
        >
          <input
            autocomplete="off"
            data-testid="input-node"
            id="rule-resource"
            name="rule-resource"
            value=""
          />
          <div
            class="message"
            data-testid="message"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`AddPolicyToRole should render fail to submit policy name is required 1`] = `
<div
  class="css-1j49gd8"
  data-testid="add-modal-message"
>
  <div
    autocomplete="off"
    class="css-vcd2ur"
    data-testid="add-rule-form"
  >
    <div
      class="css-y97ykc"
    >
      <label
        class="css-fc0mv0-input-label denali-input-label css-9xcde0"
        for="policy-name"
      >
        Policy Name
      </label>
      <div
        class="css-1xqce5"
      >
        <div
          class="css-dqae60-input denali-input animated css-fvi5z5"
          data-testid="input-wrapper"
        >
          <input
            autocomplete="off"
            data-testid="input-node"
            id="policy-name"
            name="policy-name"
            placeholder="Policy Name"
            value=""
          />
          <div
            class="message"
            data-testid="message"
          />
        </div>
      </div>
    </div>
    <div
      class="css-y97ykc"
    >
      <label
        class="css-fc0mv0-input-label denali-input-label css-9xcde0"
        for="rule-effect"
      >
        Rule Effect
      </label>
      <div
        class="css-1xqce5"
      >
        <div
          class="css-9o2u2w-button-group denali-radiobutton-group css-1ldveyy"
          data-testid="radiobuttongroup"
        >
          <div
            class="css-fe5os4-radiobutton denali-radiobutton"
            data-testid="radiobutton-wrapper"
          >
            <input
              checked=""
              id="radiobutton-rule-effect-radio-group-ALLOW"
              name="rule-effect-radio-group"
              type="radio"
              value="ALLOW"
            />
            <label
              for="radiobutton-rule-effect-radio-group-ALLOW"
            >
              Allow
            </label>
          </div>
          <div
            class="css-fe5os4-radiobutton denali-radiobutton"
            data-testid="radiobutton-wrapper"
          >
            <input
              id="radiobutton-rule-effect-radio-group-DENY"
              name="rule-effect-radio-group"
              type="radio"
              value="DENY"
            />
            <label
              for="radiobutton-rule-effect-radio-group-DENY"
            >
              Deny
            </label>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-y97ykc"
    >
      <label
        class="css-fc0mv0-input-label denali-input-label css-9xcde0"
        for="rule-action"
      >
        Rule Action
      </label>
      <div
        class="css-1xqce5"
      >
        <div
          class="css-dqae60-input denali-input animated css-fvi5z5"
          data-testid="input-wrapper"
        >
          <input
            autocomplete="off"
            data-testid="input-node"
            id="rule-action"
            name="rule-action"
            placeholder="Rule Action"
            value=""
          />
          <div
            class="message"
            data-testid="message"
          />
        </div>
      </div>
    </div>
    <div
      class="css-y97ykc"
    >
      <label
        class="css-fc0mv0-input-label denali-input-label css-9xcde0"
        for="rule-resource"
      >
        Rule Resource
      </label>
      <div
        class="css-1xqce5"
      >
        <div
          class="css-dqae60-input denali-input animated css-fvi5z5"
          data-testid="input-wrapper"
        >
          <input
            autocomplete="off"
            data-testid="input-node"
            id="rule-resource"
            name="rule-resource"
            value=""
          />
          <div
            class="message"
            data-testid="message"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`AddPolicyToRole should render failed to submit action is required 1`] = `
<div
  class="css-1j49gd8"
  data-testid="add-modal-message"
>
  <div
    autocomplete="off"
    class="css-vcd2ur"
    data-testid="add-rule-form"
  >
    <div
      class="css-y97ykc"
    >
      <label
        class="css-fc0mv0-input-label denali-input-label css-9xcde0"
        for="policy-name"
      >
        Policy Name
      </label>
      <div
        class="css-1xqce5"
      >
        <div
          class="css-dqae60-input denali-input animated css-fvi5z5"
          data-testid="input-wrapper"
        >
          <input
            autocomplete="off"
            data-testid="input-node"
            id="policy-name"
            name="policy-name"
            placeholder="Policy Name"
            value=""
          />
          <div
            class="message"
            data-testid="message"
          />
        </div>
      </div>
    </div>
    <div
      class="css-y97ykc"
    >
      <label
        class="css-fc0mv0-input-label denali-input-label css-9xcde0"
        for="rule-effect"
      >
        Rule Effect
      </label>
      <div
        class="css-1xqce5"
      >
        <div
          class="css-9o2u2w-button-group denali-radiobutton-group css-1ldveyy"
          data-testid="radiobuttongroup"
        >
          <div
            class="css-fe5os4-radiobutton denali-radiobutton"
            data-testid="radiobutton-wrapper"
          >
            <input
              checked=""
              id="radiobutton-rule-effect-radio-group-ALLOW"
              name="rule-effect-radio-group"
              type="radio"
              value="ALLOW"
            />
            <label
              for="radiobutton-rule-effect-radio-group-ALLOW"
            >
              Allow
            </label>
          </div>
          <div
            class="css-fe5os4-radiobutton denali-radiobutton"
            data-testid="radiobutton-wrapper"
          >
            <input
              id="radiobutton-rule-effect-radio-group-DENY"
              name="rule-effect-radio-group"
              type="radio"
              value="DENY"
            />
            <label
              for="radiobutton-rule-effect-radio-group-DENY"
            >
              Deny
            </label>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-y97ykc"
    >
      <label
        class="css-fc0mv0-input-label denali-input-label css-9xcde0"
        for="rule-action"
      >
        Rule Action
      </label>
      <div
        class="css-1xqce5"
      >
        <div
          class="css-dqae60-input denali-input animated css-fvi5z5"
          data-testid="input-wrapper"
        >
          <input
            autocomplete="off"
            data-testid="input-node"
            id="rule-action"
            name="rule-action"
            placeholder="Rule Action"
            value=""
          />
          <div
            class="message"
            data-testid="message"
          />
        </div>
      </div>
    </div>
    <div
      class="css-y97ykc"
    >
      <label
        class="css-fc0mv0-input-label denali-input-label css-9xcde0"
        for="rule-resource"
      >
        Rule Resource
      </label>
      <div
        class="css-1xqce5"
      >
        <div
          class="css-dqae60-input denali-input animated css-fvi5z5"
          data-testid="input-wrapper"
        >
          <input
            autocomplete="off"
            data-testid="input-node"
            id="rule-resource"
            name="rule-resource"
            value=""
          />
          <div
            class="message"
            data-testid="message"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
